libs/cardano-protocol-tpraos/test/Test/Cardano/Protocol/Binary/CddlSpec.hs

```diff
@@ -1,46 +1,68 @@
 import Cardano.Protocol.TPraos.BHeader (BHBody, BHeader)
 import Cardano.Protocol.TPraos.OCert (OCert)
+import Codec.CBOR.Cuddle.Huddle (Huddle)
 import qualified Data.ByteString.Lazy as BSL
-import Test.Cardano.Ledger.Binary.Cddl (
-  beforeAllCddlFile,
-  cddlDecoderEquivalenceSpec,
-  cddlRoundTripAnnCborSpec,
-  cddlRoundTripCborSpec,
- )
+import Test.Cardano.Ledger.Allegra.Binary.Cddl (readAllegraCddlFiles)
+import Test.Cardano.Ledger.Allegra.CDDL (allegraCDDL)
+import Test.Cardano.Ledger.Alonzo.Binary.Cddl (readAlonzoCddlFiles)
+import Test.Cardano.Ledger.Alonzo.CDDL (alonzoCDDL)
+import Test.Cardano.Ledger.Binary.Cddl (beforeAllCddlFile, cddlRoundTripCborSpec)
 import Test.Cardano.Ledger.Binary.Cuddle
 import Test.Cardano.Ledger.Common
+import Test.Cardano.Ledger.Mary.Binary.Cddl (readMaryCddlFiles)
+import Test.Cardano.Ledger.Mary.CDDL (maryCDDL)
 import Test.Cardano.Ledger.Shelley.Binary.Cddl (readShelleyCddlFiles)
 import Test.Cardano.Ledger.Shelley.CDDL (shelleyCDDL)
+import Test.Cardano.Protocol.Binary.Cddl (
+  cddlEraSpec,
+  huddleEraSpec,
+ )
 
 spec :: Spec
 spec =
   describe "CDDL" $ do
     let n = 3
-    specForEra @ShelleyEra readShelleyCddlFiles n
+    specForEra @ShelleyEra readShelleyCddlFiles shelleyCDDL n
+    specForEra @AllegraEra readAllegraCddlFiles allegraCDDL n
+    specForEra @MaryEra readMaryCddlFiles maryCDDL n
+    specForEra @AlonzoEra readAlonzoCddlFiles alonzoCDDL n
 
-specForEra :: forall era. Era era => IO [BSL.ByteString] -> Int -> Spec
-specForEra readCddlFiles n = do
+specForEra ::
+  forall era.
+  (Era era, AtMostEra AlonzoEra era) =>
+  IO [BSL.ByteString] ->
+  Huddle ->
+  Int ->
+  Spec
+specForEra readCddlFiles cddlFiles n = do
   describe (eraName @era) $ do
-    let v = eraProtVerLow @era
-    beforeAllCddlFile n readCddlFiles $ do
-      cddlRoundTripAnnCborSpec @(BHeader StandardCrypto) v "header"
-      cddlRoundTripCborSpec @(BHeader StandardCrypto) v "header"
-      cddlRoundTripCborSpec @(BHBody StandardCrypto) v "header_body"
-      cddlRoundTripCborSpec @(CBORGroup (OCert StandardCrypto)) v "[ operational_cert ]"
-      -- TODO: add Huddle round trip tests
-      describe "DecCBOR instances equivalence via CDDL" $ do
-        cddlDecoderEquivalenceSpec @(BHeader StandardCrypto) v "header"
-    describe "DecCBOR instances equivalence via CDDL - Huddle" $ do
-      specWithHuddle shelleyCDDL 100 $ do
-        huddleDecoderEquivalenceSpec @(BHeader StandardCrypto) v "header"
+    describe "Ruby-based" $
+      beforeAllCddlFile n readCddlFiles $ do
+        cddlEraSpec @era @StandardCrypto @BHeader @BHBody
+        cddlRoundTripCborSpec @(CBORGroup (OCert StandardCrypto))
+          (eraProtVerLow @era)
+          "[ operational_cert ]"
+
+    describe "Huddle" $
+      specWithHuddle cddlFiles 100 $ do
+        huddleEraSpec @era @StandardCrypto @BHeader @BHBody
+        xdescribe "Cannot generate a CBOR term corresponding to a group with cuddle" $
```

Comment:
@nc6  -  Could you confirm if this is by design like this, or if it's just something pending atm in cuddle?